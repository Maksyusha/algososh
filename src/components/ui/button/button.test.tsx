import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("button component", () => {
  it("renders with text", () => {
    const tree = renderer.create(<Button text="test"></Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders without text", () => {
    const tree = renderer.create(<Button></Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with disabled state", () => {
    const tree = renderer.create(<Button disabled></Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with loading state", () => {
    const tree = renderer.create(<Button isLoader={true}></Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("callback", () => {
    const callback = jest.fn();
    render(<Button onClick={callback}></Button>);

    fireEvent.click(screen.getByRole("button"));
    expect(callback).toHaveBeenCalled();
  });
});
