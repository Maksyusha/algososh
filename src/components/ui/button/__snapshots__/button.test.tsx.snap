// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component renders with disabled state 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button component renders with loading state 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button component renders with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`button component renders without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
